frontend/src/metabase/query_builder/components/filters/FilterPopover/FilterPopover.tsx

```diff
@@ -146,8 +146,7 @@ export default function FilterPopover({
   };
 
   const handleFilterChange = (mbql: any[] = []) => {
-    const newFilter = new Filter(mbql, filter?.index(), query);
-
+    const newFilter = filter ? filter.set(mbql) : new Filter(mbql, null, query);
```

Comment:
Here how it looked like in 0.43 https://github.com/metabase/metabase/blob/c5aebf70dc164bde6d2247c0bf1762a4203f9d9a/frontend/src/metabase/query_builder/components/filters/FilterPopover.tsx#L141

Reply:
The root cause is that `filter.set` uses the query for the same stage level, while `query` here is the top-level query.